include/deal.II/grid/tria_accessor.templates.h

```diff
@@ -2039,6 +2039,71 @@ TriaAccessor<structdim, dim, spacedim>::diameter () const
 }
 
 
+template <int structdim, int dim, int spacedim>
+std::pair<Point<spacedim>,double>
+TriaAccessor<structdim, dim, spacedim>::smallest_enclosing_sphere () const
+{
+  switch (structdim)
+    {
+    case 1:
+      const Point<spacedim> p10 = this->vertex(1)-this->vertex(0);
+      return std::make_pair(this->vertex(0)+0.5*p10, p10.norm());
+    case 2:
+      const Point<spacedim> p30 = this->vertex(3)-this->vertex(0);
+      const Point<spacedim> p21 = this->vertex(2)-this->vertex(1);
+
+      const double diameter =0.;
+      const Point<spacedim> center;
+
+      (p30.norm() > p21.norm()) ?
+      (
+        center   = this->vertex(0) + 0.5*p30,
+        diameter = p30.norm()
+      )
+      :
+      (
+        center   = this->vertex(1) + 0.5*p21,
+        diameter = p21.norm()
+      );
+
+      return std::make_pair ( center, diameter);
+
+    case 3:
+      const Point<spacedim> p70 = this->vertex(7)-this->vertex(0);
+      const Point<spacedim> p61 = this->vertex(6)-this->vertex(1);
+      const Point<spacedim> p25 = this->vertex(2)-this->vertex(5);
+      const Point<spacedim> p34 = this->vertex(3)-this->vertex(4);
+
+      std::vector<double> diagonals= { p70.norm(), p61.norm(), p25.norm(), p34.norm()};
+
+      const std::vector<double>::iterator it = std::max_element( diagonals.begin(), diagonals.end() );
+
+      const double diameter = 0.;
+      const Point<spacedim> center;
+
+      ( it == diagonals.begin())
+      ?
+      ( diameter = *it, center = this->vertex(0) + 0.5*Point<spacedim> p70 )
+      :
+      (( it == diagonals.begin()+1)
+       ?
+       ( diameter = *it, center = this->vertex(1) + 0.5*Point<spacedim> p61 )
+       :
+       (( it == diagonals.begin()+2)
+        ?
+        ( diameter = *it, center = this->vertex(5) + 0.5*Point<spacedim> p25 )
+        :
+        ( diameter = *it, center = this->vertex(4) + 0.5*Point<spacedim> p34 )
+       )
+      );
+
+      return std::make_pair(center, diameter);
+    default:
+      Assert (false, ExcNotImplemented());
+      return -1e10;
```

Comment:
Since this shouldn't happen, instead of `-1e10`, use `numbers::signaling_NaN<double>()`.